gradebook.cpp

```diff
@@ -137,17 +139,104 @@ double Gradebook::get_category_grade(std::string category){
         return this->Get_Exam_Grade();
     }
     else {
-	    return -1.0;
+        return -1.0;
     }
     return 0;
 }
 
 // Return the total cumulative grade for the class as a percentage
 double Gradebook::get_total_grade(){
-    return 0;
+    int tot_points = 1000;
+    double tot_grade = this->Get_Assignment_Grade()+ this->Get_Exam_Grade() + this->Get_Lab_Grade() + this->Get_Projects_Grade();
+    return (tot_grade/tot_points)*100;
 }
 
 // Manually add a new grade to the respective category
-void Gradebook::add_grade(std::string category,std::string name, double grade){
-
-}
+void Gradebook::add_grade(std::string category,std::string name, double grade,std::string fname){
+    if(category == LABS){
+        for(int i = 0; i < labs.size();i++){
+            if(std::stod(name) < std::stod(labs[i].first)){
+                labs.insert(labs.begin()+i,{name,grade});
+                i = labs.size();
+            }
+            else if(i == labs.size()-1){
+                labs.push_back({name,grade});
+                i = labs.size();
+            }
+        }
+        this->lab_grades_total += grade;
+    }
+    else if(category == ASSIGNMENTS){
+        for(int i = 0; i < assignments.size();i++){
+            if(std::stod(name) < std::stod(assignments[i].first)){
+                assignments.insert(assignments.begin()+i,{name,grade});
+                i = assignments.size();
+            }
+            else if(i == assignments.size()-1){
+                assignments.push_back({name,grade});
+                i = assignments.size();
+            }
+        }
+        this->assignment_grades_total+= grade;
+    }
+    else if(category == PROJECTS){
+        for(int i = 0; i < projects.size();i++){
+            if(std::stod(name) < std::stod(projects[i].first)){
+                projects.insert(projects.begin()+i,{name,grade});
+                i = projects.size();
+            }
+            else if(i == projects.size()-1){
+                projects.push_back({name,grade});
+                i = projects.size();
+            }
+        }
+        this->project_grades_total += grade;
+    }
+    else if(category == EXAMS){
+        for(int i = 0; i < exams.size();i++){
+            if(std::stod(name) < std::stod(exams[i].first)){
+                exams.insert(exams.begin()+i,{name,grade});
+                i = projects.size();
+            }
+            else if(i == exams.size()-1){
+                exams.push_back({name,grade});
+                i = projects.size();
+            }
+        }
+        this->exam_grades_total += grade;
+    }
+    std::ofstream new_file(fname);
+    bool lab_done = false;
+    bool assignment_done = false;
+    bool proj_done = false;
+    bool exam_done = false;
+    if(new_file.is_open()){

```

Comment:
I also don't think that a new file should be created here. We should have a function that does that once the program is being closed. It would be unnecessary and inefficient to rewrite the whole file every time a new grade was manually added.

Reply:
That's fine, I didn't know how we wanted to approach it so I figured we could just copy and paste it